---
title: "VarCovar Rembrandt"
author: "TF"
date: "`r Sys.Date()`"
output:
  bookdown::github_document2:
    toc: True
always_allow_html: true
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE, message = FALSE, warning = FALSE)
```

```{r}
library(dplyr)
library(ggplot2)

source(file = "functions/fct_report_simu_setting.R")

dfRembrandt <- readRDS("data/rembrandt/dfresvarcovar.rds")
dfBreast <- readRDS("data/breast_cancer/dfresvarcovar.rds")
```

# Introduction

Document is separated into two parts

- First part goal is to find relevant simulation setting from Rembrandt and Breast cancer study.
- Second part will present the results from the simulation studies

# Finding the simulation setting

We used Rembrandt and breast cancer datasets. For each datasets we explored the parameters of interest to perform the simulation study.

Each dot in the following figure corresponds to one pathway.

## Proportion of significant genes

```{r}
plot_R <- fct_prop_sign_genes(dfRembrandt) + ggtitle("Rembrandt")
plot_B <- fct_prop_sign_genes(dfBreast) + ggtitle("Breast")

ggpubr::ggarrange(plot_R, plot_B)
```

Choices :

- number of genes : $NG = \{10, 50, 200\}$
- proportion of significant genes : $p_g = \{0.2, 0.5\}$

## Beta from Cox model

```{r}
lsbeta_R <- fct_beta_dist(dfRembrandt, caption = "Rembrandt : Proportion of positive, negative and non significant genes")
lsbeta_B <- fct_beta_dist(dfBreast, caption = "Breast cancer : Proportion of positive, negative and non significant genes")

lsbeta_R$table_beta
lsbeta_B$table_beta
```

Proportions are roughly similar between positive and negative genes.

```{r}
plot_R <- lsbeta_R$plot_unique_beta + ggtitle("Rembrandt")
plot_B <- lsbeta_B$plot_unique_beta + ggtitle("Breast")

ggpubr::ggarrange(plot_R, plot_B)
```

```{r}
plot_R <- lsbeta_R$plot_multiple_beta + ggtitle("Rembrandt")
plot_B <- lsbeta_B$plot_multiple_beta + ggtitle("Breast")

ggpubr::ggarrange(plot_R, plot_B)
```

Choices :

- $\beta \sim \mathcal{N}(0, 0.4^2)$
- half of genes follows $\beta \sim \mathcal{N}(-0.4, 0.2^2)$ and half follows $\beta \sim \mathcal{N}(0.4, 0.2^2)$
- half of genes follows $\beta \sim \mathcal{N}(-0.8, 0.4^2)$ and half follows $\beta \sim \mathcal{N}(0.8, 0.4^2)$

## Gene expression variance

```{r}
plot_R <- fct_plot_variance(dfRembrandt) + ggtitle("Rembrandt")
plot_B <- fct_plot_variance(dfBreast, nb_filter_plot = 3) + ggtitle("Breast")

ggpubr::ggarrange(plot_R, plot_B)
```

Choice :

- Variance : $C_{jj} = 0.2$

## Gene expression correlation

### Overall gene expression correlation

```{r}
ls_R <- fct_plot_correlation(dfRembrandt, caption = "Rembrandt : Shapes of non-standard beta distribution by type of gene")
ls_B <- fct_plot_correlation(dfBreast, caption = "Breast : Shapes of non-standard beta distribution by type of gene")

plot_R <- ls_R$plot_unique + ggtitle("Rembrandt")
plot_B <- ls_B$plot_unique + ggtitle("Breast")

ggpubr::ggarrange(plot_R, plot_B)
```


```{r}
ls_R$table_shapes_correlation
ls_B$table_shapes_correlation
```

### Gene expression correlation depending on gene significance

```{r}
plot_R <- ls_R$plot_multiple + ggtitle("Rembrandt")
plot_B <- ls_B$plot_multiple + ggtitle("Breast")

ggpubr::ggarrange(plot_R, plot_B)
```

Choice :

- Overall correlation follows a non-standard beta $corr \sim NSBeta(20, 20, min=-1, max =1)$
- Correlation between significant genes $corr \sim NSBeta(10, 10, min=-1, max =1)$ and between other genes $corr \sim NSBeta(25, 25, min=-1, max =1)$

## Summary Rembrandt and simulation settings

Taking paper notations :

- number of genes : $NG = \{10, 50, 200\}$
- number of individuals : $N = \{50, 100\}$
- proportion of significant genes : $p_g = \{0.2, 0.5\}$
- proportion of censored genes : $c = \{0, 0.3\}$
- Variance : $C_{jj} = 0.2$

<!-- **Note : scenario D, E and F do not rely on correlation matrix to simulate variance covariance matrix so the those matrix are unlikely to be positive definite.** -->

### Case

- Case 4 : Overall correlation follows a non-standard beta $corr \sim NSBeta(20, 20, min=-1, max =1)$
- Case 5 : Correlation between significant genes $corr \sim NSBeta(10, 10, min=-1, max =1)$ and between other genes $corr \sim NSBeta(25, 25, min=-1, max =1)$

**Note : Case IV and V are not guaranteed to generate positive definite matrix, after the correlation values are sampled, if the matrix is not positive definite, the nearest positive definite matrix is computed using the algorithm defined by Higham 2002 (see Matrix::nearPD). This correction was not implemented for scenario D, E and F.**

- Case 6 : Correlation between significant genes is $0.2$, correlation with non significant genes is $0$.

### Type

- Type (D) : $\beta \sim \mathcal{N}(0, 0.4^2)$
- Type (E) : half of genes follows $\beta \sim \mathcal{N}(-0.4, 0.2^2)$ and half follows $\beta \sim \mathcal{N}(0.4, 0.2^2)$
- Type (F) : half of genes follows $\beta \sim \mathcal{N}(-0.8, 0.4^2)$ and half follows $\beta \sim \mathcal{N}(0.8, 0.4^2)$
- Type (G) : $\beta \sim \mathcal{N}(0, 0.4^2)$. Correlation matrix of $\beta$ is the same as correlation matrix of genes.
- Type (H) : half of genes follows $\beta \sim \mathcal{N}(-0.4, 0.2^2)$ and half follows $\beta \sim \mathcal{N}(0.4, 0.2^2)$. Correlation matrix of $\beta$ is the same as correlation matrix of genes.
- Type (I) : half of genes follows $\beta \sim \mathcal{N}(-0.8, 0.4^2)$ and half follows $\beta \sim \mathcal{N}(0.8, 0.4^2)$. Correlation matrix of $\beta$ is the same as correlation matrix of genes.

# Results of simulation study

```{r fig.height=10, fig.cap="Statistical power depending on simulation setting. N is the number of patients and NG the number of genes in the pathway. Note : scenario D, E and F do not rely on correlation matrix to simulate variance covariance matrix so the those matrix are unlikely to be positive definite."}
source("script/script_simulation/figures_simu_rembrandt_breast_v1.R")

plot_simu
```

```{r fig.height=14, fig.cap="Statistical power depending on simulation setting (second batch). N is the number of patients and NG the number of genes in the pathway."}
source("script/script_simulation/figures_simu_rembrandt_breast_v2.R")

plot_simu
```

```{r fig.height=6, fig.cap="Time depending on simulation setting (second batch). N is the number of patients and NG the number of genes in the pathway."}
plot_time
```

# Discussion

Overall :

- Global Test is fast compared to other approach. It is not completely clear to me why because it also relies on permutation and is similar to Wald test. There is probably a clever implementation ?
- In terms of power Wald test is the best option, closely followed by Global test
- sGBJ is slow when the number of genes rise. This is due to permutation for variance matrix estimation that might be reduced to improve computation time.
- sGBJ performance is similar to most approach except when the number of genes increases.

## Comprehend tests

see : https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3196970/

### Wald test

- estimate one beta by gene using cox model
- sum the wald test statistics

### Global test

Jelle J. Goeman and others, Testing association of a pathway with survival using gene expression data, Bioinformatics, Volume 21, Issue 9, May 2005, Pages 1950â€“1957, https://doi.org/10.1093/bioinformatics/bti267

- estimate one beta by gene using cox model
- makes the assumption that beta are centered on 0
- H0 : variance of beta is equal to 0.

### Global boost test

Compute the cox model negative likelihood using a boosted multivariate (all genes) cox model. P-value is computed by permutation on the negative likelihood.

### Thinking

So basically, Wald and Global are super similar, Global test would have been expected to perform better in D case because the hypothesis on beta distribution is closer. One explanation could be that this test is optimal for small deviations from null hypothesis. This might explain why is seems better than Wald where power is low and why Wald is better when power is high.

Global boost test relies on one model for all the data contrary to other models that perform one model by gene which might makes it more adapted to large number of genes. This is partially in line with good performance when N = 50.

sGBJ is similar to Wald and Global test in the sense that it is based on one cox model by feature. However, it takes into account the covariance of genes in its formula so it should perform better when beta are correlated.
